clients/unshaded/src/main/java/tachyon/client/next/block/BlockWorkerClientPool.java

```diff
@@ -39,14 +40,18 @@
   private final TachyonConf mTachyonConf;
 
   public BlockWorkerClientPool(NetAddress workerNetAddress, TachyonConf conf) {
-    // TODO: Get capacity from conf
-    super(10000);
-    mExecutorService = Executors.newFixedThreadPool(10, ThreadFactoryUtils.build(
-        "block-worker-heartbeat-%d", true));
+    super(getBlockWorkerClientTotalThreads(conf));
+    int blockWorkerClientThreads = conf.getInt(Constants.USER_BLOCK_WORKER_CLIENT_THREADS);
+    mExecutorService = Executors.newFixedThreadPool(blockWorkerClientThreads,
+        ThreadFactoryUtils.build("block-master-heartbeat-%d", true));
     mWorkerNetAddress = workerNetAddress;
     mTachyonConf = conf;
   }
 
+  private static int getBlockWorkerClientTotalThreads(TachyonConf conf) {
```

Comment:
because the call to super class' constructor cannot be preceded by a call to retrieve the conf (or any other calls)